src/sentry/api/serializers/models/notification_setting.py

```diff
@@ -35,22 +37,31 @@ def get_attrs(
         """
         type_option: Optional[NotificationSettingTypes] = kwargs.get("type")
 
-        # TODO(actorid) This needs to be reworked. `item_list` can either be
-        # a list of Team or User instances. Each item type requires different
-        # query logic for getting notification settings.
-        actor_mapping = {recipient.actor_id: recipient for recipient in item_list}
+        team_map = {
+            t.id: t for t in list(filter(lambda x: isinstance(x, (Team, RpcTeam)), item_list))
+        }
+        user_map = {
+            u.id: u for u in list(filter(lambda x: isinstance(x, (User, RpcUser)), item_list))
+        }
```

Comment:
I believe this can be simplified to

```suggestion
        team_map = {t.id: t for t in item_list if isinstance(t, (Team, RpcTeam))}
        user_map = {u.id: u for u in item_list if isinstance(u, (User, RpcUser))}
```